packages/core/src/core/index.ts

```diff
@@ -121,4 +121,26 @@ export const Auth = {
       () => null
     );
   },
+
+  validateSessionAndGetProfile: async function (
```

Comment:
maybe we should have a single function to do everything.
If only the getProfile fails, we might not want to fails everything to track the user.
Without a single function it becomes easier to handle such a case.